errors.go

```diff
@@ -215,6 +235,36 @@ func (errorType ErrorType) UserErrorf(msg string, args ...interface{}) error {
 	}
 }
 
+// AddDetails adds a details element to an error
+func (errorType ErrorType) AddDetails(err error, details interface{}) error {
+	if err == nil {
+		return errorType.Details(details)
```

Comment:
I'm having trouble replying to @tzvatot, but I align with what he says. Let's keep it that way

Reply:
But let's make `Details` private or inline it here.
I don't want to encourage creating details with no error message

Reply:
I'm fine with making it private. Done.